### PR TITLE
Fix/error boundary

### PR DESCRIPTION
**Pull request by sanjayankur31**
Don't use the `ErrorDialog` at all, propagate errors to the `ErrorBoundary` instead.

### CODE REVIEW

**Comment by sanjayankur31** [comment]
I noticed that the osbbackend is the only place using `setError`---can we/should we tweak that so that the error is propagated to the front-end and caught by the ErroBoundary too?

**Comment by sanjayankur31** [comment]
Example private workspace to test with, without signing in:

https://localhost:3000/workspace/84

**Review by filippomc — CHANGES_REQUESTED** [review]
Not sure about the message, the error at this level may be any.
Also I think we can improve this by showing directly a modal on top of the children components instead of showing a blank page (before it was implemented the simple way as it was supposed to catch only runtime errors)

File: applications/osb-portal/src/components/sentry/SentryErrorBoundary.tsx [hunk 1]
```diff
@@ -38,7 +40,17 @@ class SentryErrorBoundary extends React.Component<{}, OwnState> {
     if (this.state.hasError) {
       // render fallback UI
       return (
-        <Button onClick={this.showDialog}>Report feedback</Button>
+        <>
+          <Box textAlign='center'>
+            <Box padding={2}>
+              <Typography align="center" display="block" variant="h4" color="textPrimary">Oops. Something went wrong. This page could not be found.</Typography>
```
[the remark this answers is not in the document]
Comment (sanjayankur31) [reply applications/osb-portal/src/components/sentry/SentryErrorBoundary.tsx]: Sure, left it at "Oops. Something went wrong".

**Comment by sanjayankur31** [comment]
> Not sure about the message, the error at this level may be any. Also I think we can improve this by showing directly a modal on top of the children components instead of showing a blank page (before it was implemented the simple way as it was supposed to catch only runtime errors)

So the sentry thing has it's own dialog already which should be shown when "report feedback" is clicked. I don't currently know why this doesn't work. It can also be automatically thrown from the looks of it using the `showDialog` etc. options:

https://docs.sentry.io/platforms/javascript/guides/react/components/errorboundary/

I did try to use the ErrorDialog here instead of the boring text before I removed it completely, but I ran into issues. For one, the error boundary is outside the theme (right?) so the dialog is not themed when used in this sentryerrorboundary bit. How does one make the theme available to whatever the error boundary wants to render, switch the `ThemeProvider` and `SentryErrorBoundary` in `App.tsx`? Or do we need to theme this other dialog separately?

Next, even if we use a dialog, in this example case the workspace page doesn't seem to even load partially for me. so I don't know if the other components of the workspace page are being rendered or not. I just get the error in the dev tools:

```
The above error occurred in the <WorkspacePage> component:

WorkspacePage@webpack-internal:///./src/pages/WorkspacePage.tsx:160:21
ConnectFunction@webpack-internal:///./node_modules/react-redux/es/components/connectAdvanced.js:231:68
Route@webpack-internal:///./node_modules/react-router/esm/react-router.js:464:29
Switch@webpack-internal:///./node_modules/react-router/esm/react-router.js:670:29
div
Router@webpack-internal:///./node_modules/react-router/esm/react-router.js:93:30
BrowserRouter@webpack-internal:///./node_modules/react-router-dom/esm/react-router-dom.js:59:35
ThemeProvider@webpack-internal:///./node_modules/@material-ui/styles/esm/ThemeProvider/ThemeProvider.js:44:18
SentryErrorBoundary@webpack-internal:///./src/components/sentry/SentryErrorBoundary.tsx:32:9
App@webpack-internal:///./src/App.tsx:47:21
ConnectFunction@webpack-internal:///./node_modules/react-redux/es/components/connectAdvanced.js:231:68
Provider@webpack-internal:///./node_modules/react-redux/es/components/Provider.js:16:15

React will try to recreate this component tree from scratch using the error boundary you provided, SentryErrorBoundary. react-dom.development.js:20085
    React 18
    renderMain index.tsx:17
```

I even commented out the components from the sentry render bit to just `return ()`, and I still get just a blank page.

---------

If a themed dialog can be used, that'll be great, but then we won't want our own dialog and the sentry dialog too. We should just use one (either a new one, or customise the sentry one: https://docs.sentry.io/platforms/javascript/guides/react/enriching-events/user-feedback/#customizing-the-widget )?

**Comment by sanjayankur31** [comment]
Sounds good. @pgleeson and I had a chat and while we are happy with errors going to Sentry, we also do want to handle general errors in the components themselves so that we can provide users with some feedback about what may have gone wrong.

So:

- handle errors in components themselves (for example when the workspace service fails) and show an error dialog there to the user about the error
- for other "system errors" that we don't (or can't) handle, let Sentry do its thing

How does that sound?

And yeh, I see that we defined our own error boundary called `SentryErrorBoundary` that seems to be in use. I'll let you make your tweaks before making any more changes here I guess.

**Comment by sanjayankur31** [comment]
(So let's not merge this PR in the meantime XD)

**Comment by filippomc** [comment]
> 

@sanjayankur31 Did the tweaks: https://github.com/OpenSourceBrain/OSBv2/pull/385/commits/99cac52431a025039beba75391b648652ea93064

This works only in the WorkspacePage, so need to to the same for other pages.

**Comment by sanjayankur31** [comment]
OK, cool, thanks! Let me pull all these changes in and update the other pages.

**Comment by sanjayankur31** [comment]
@filippomc works very well. I tweaked the pages a little to pass errors to the boundary.

If this looks OK, good to merge now.

**Review by filippomc — APPROVED** [review]
Good to go

**Pull request merged**